### PR TITLE
Add k8s 1.31 test for cert-manager 1.15

### PR DESCRIPTION
**Pull request by SgtCoDFish**
Follows on from https://github.com/cert-manager/cert-manager/pull/7305

### CODE REVIEW

**Review by wallrj — APPROVED** [review]
A couple of  non-blocking questions and a suggestion.

/approve
/lgtm

/hold in case you want to address those here.

File: config/prowgen/prowspecs/specs.go [hunk 1]
```diff
@@ -109,7 +109,7 @@ var knownBranches map[string]BranchSpec = map[string]BranchSpec{
 		primaryKubernetesVersion: "1.30",
 
 		// TODO: test k8s 1.31 here when possible; requires support in the release-1.15 branch on cert-manager
-		otherKubernetesVersions: []string{"1.25", "1.26", "1.27", "1.28", "1.29"},
+		otherKubernetesVersions: []string{"1.25", "1.26", "1.27", "1.28", "1.29", "1.31"},
```
Comment (wallrj) [inline config/prowgen/prowspecs/specs.go:112]: Ah, I'd forgotten that all the YAML is generated  these days.
Perhaps that's why the cron schedules changed.

**Comment by wallrj** [comment]
/unhold

I'll fix that comment in https://github.com/cert-manager/testing/pull/1061

File: config/prowgen/prowspecs/specs.go (hunk 1 above)
Comment (SgtCoDFish) [reply config/prowgen/prowspecs/specs.go:112]: It is, yeah!

**Pull request merged**